novawallet/Modules/Vote/Governance/Operation/Action/Gov2ActionOperationFactory.swift

```diff
@@ -8,27 +8,51 @@ final class Gov2ActionOperationFactory: GovernanceActionOperationFactory {
         requestFactory: StorageRequestFactoryProtocol,
         connection: JSONRPCEngine,
         codingFactoryOperation: BaseOperation<RuntimeCoderFactoryProtocol>
-    ) -> CompoundOperationWrapper<[StorageResponse<Preimage.RequestStatus>]> {
-        OperationCombiningService.compoundNonOptionalWrapper(
+    ) -> CompoundOperationWrapper<Preimage.RequestStatus?> {
+        let fetchOperation = OperationCombiningService(
             operationManager: OperationManager(operationQueue: operationQueue)
         ) {
             let codingFactory = try codingFactoryOperation.extractNoCancellableResultData()
 
-            let storagePath: StorageCodingPath
+            var wrappers: [CompoundOperationWrapper<[StorageResponse<Preimage.RequestStatus>]>] = []
 
             if codingFactory.hasStorage(for: Preimage.requestStatusForStoragePath) {
-                storagePath = Preimage.requestStatusForStoragePath
-            } else {
-                storagePath = Preimage.statusForStoragePath
+                wrappers.append(
+                    requestFactory.queryItems(
+                        engine: connection,
+                        keyParams: { [BytesCodable(wrappedValue: hash)] },
+                        factory: { codingFactory },
+                        storagePath: Preimage.requestStatusForStoragePath
+                    )
+                )
             }
 
-            return requestFactory.queryItems(
-                engine: connection,
-                keyParams: { [BytesCodable(wrappedValue: hash)] },
-                factory: { codingFactory },
-                storagePath: storagePath
-            )
+            if codingFactory.hasStorage(for: Preimage.statusForStoragePath) {
+                wrappers.append(
+                    requestFactory.queryItems(
+                        engine: connection,
+                        keyParams: { [BytesCodable(wrappedValue: hash)] },
+                        factory: { codingFactory },
+                        storagePath: Preimage.statusForStoragePath
+                    )
+                )
+            }
+
+            return wrappers
```

Comment:
It can be reduced to:
let storages = [requestStatusForStoragePath, statusForStoragePath]
wrappers = storages.compactMap {
       codingFactory.hasStorage(for: $0) ? requestFactory.queryItems... : nil
}
or
let storages = [requestStatusForStoragePath, statusForStoragePath].filter(codingFactory.hasStorage)
wrappers = storages.map

Reply:
Good catch. Will update in the next PRs